src/bootloaders/systemd-class.c

```diff
@@ -384,6 +384,46 @@ bool sd_class_set_default_kernel(const BootManager *manager, const Kernel *kerne
         return true;
 }
 
+char *sd_class_get_default_kernel(const BootManager *manager)
+{
+        if (!manager) {
+                return NULL;
+        }
+
+        autofree(char) *conf = NULL;
+        autofree(char) *fstring = NULL;
+        char ktype[32] = { 0 };
+        char version[16] = { 0 };
+        int release = -1;
+        int r = -1;
+        const char *prefix = NULL;
+
+        if (file_get_text(sd_class_config.loader_config, &conf)) {
+                char *c = strstr(conf, "default ");
+                if (!c) {
+                        return NULL;
+                }
+                errno = 0;
+                prefix = boot_manager_get_vendor_prefix((BootManager *)manager);
+                fstring = string_printf("default %s-%%31[^-]-%%15[^-]-%%d", prefix);
```

Comment:
It does not seem to be right. I'm not sure `sscanf` is capable of parsing `default Clear-linux-pk414-standard-4.14.74-115' in this form which, in turn, makes it impossible to display the kernel correctly afterwards (or grab its version information):

```
$ sudo ./clr-boot-manager list-kernels
*org.clearlinux.native.4.18.16-650
org.clearlinux.pk414-standard.4.14.74-115
$ sudo ./clr-boot-manager set-kernel org.clearlinux.pk414-standard.4.14.74-115
$ sudo ./clr-boot-manager list-kernels
org.clearlinux.native.4.18.16-650
org.clearlinux.pk414-standard.4.14.74-115
```

Because scanning "Clear-linux-pk414-standard-4.14.74-115" yields "pk414-standard-4" for `ktype`.

Reply:
Alrighty, horrible manual parsing change to make this work and test added to validate ktypes with dashes.